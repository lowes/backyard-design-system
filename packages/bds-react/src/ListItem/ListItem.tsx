import * as React from 'react'
import { ListItemWrapper } from './styles/ListItemWrapper'
import type { BackyardBaseProps } from '../utils/typings/BackyardProps'
import type { BDSForwardRef } from '../utils/typings/BDSComponentProps'

const ListItem: BDSForwardRef<
    ListItemProps & {
        ref?: any
    }
> = React.forwardRef<ListItemRef, ListItemProps>(function ListItem({ children, ...props }, ref) {
    return (
        <ListItemWrapper ref={ref} {...props}>
            {children}
        </ListItemWrapper>
    )
})

type ListItemRef = HTMLLIElement

interface ListItemProps extends BackyardBaseProps<ListItemRef> {}

ListItem.bdsName = 'ListItem'

export { ListItem }

export type { ListItemRef, ListItemProps }

export default ListItem
