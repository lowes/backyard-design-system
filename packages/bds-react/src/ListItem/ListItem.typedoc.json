{
	"id": 0,
	"name": "@lowes-tech/bds-react/ListItem",
	"kind": 1,
	"kindString": "Project",
	"flags": {},
	"children": [
		{
			"id": 7,
			"name": "default",
			"kind": 16777216,
			"kindString": "Reference",
			"flags": {},
			"target": 1
		},
		{
			"id": 5,
			"name": "ListItemProps",
			"kind": 256,
			"kindString": "Interface",
			"flags": {},
			"children": [
				{
					"id": 6,
					"name": "as",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"type": {
						"type": "reference",
						"typeArguments": [
							{
								"type": "intrinsic",
								"name": "any"
							}
						],
						"name": "ElementType"
					},
					"inheritedFrom": {
						"type": "reference",
						"name": "BackyardBaseProps.as"
					}
				}
			],
			"groups": [
				{
					"title": "Properties",
					"kind": 1024,
					"children": [
						6
					]
				}
			],
			"extendedTypes": [
				{
					"type": "reference",
					"typeArguments": [
						{
							"type": "reference",
							"id": 4,
							"name": "ListItemRef"
						}
					],
					"name": "BackyardBaseProps"
				}
			]
		},
		{
			"id": 4,
			"name": "ListItemRef",
			"kind": 4194304,
			"kindString": "Type alias",
			"flags": {},
			"type": {
				"type": "reference",
				"name": "HTMLLIElement"
			}
		},
		{
			"id": 1,
			"name": "ListItem",
			"kind": 32,
			"kindString": "Variable",
			"flags": {},
			"type": {
				"type": "reference",
				"typeArguments": [
					{
						"type": "intersection",
						"types": [
							{
								"type": "reference",
								"id": 5,
								"name": "ListItemProps"
							},
							{
								"type": "reflection",
								"declaration": {
									"id": 2,
									"name": "__type",
									"kind": 65536,
									"kindString": "Type literal",
									"flags": {},
									"children": [
										{
											"id": 3,
											"name": "ref",
											"kind": 1024,
											"kindString": "Property",
											"flags": {
												"isOptional": true
											},
											"type": {
												"type": "intrinsic",
												"name": "any"
											}
										}
									],
									"groups": [
										{
											"title": "Properties",
											"kind": 1024,
											"children": [
												3
											]
										}
									]
								}
							}
						]
					}
				],
				"name": "BDSForwardRef"
			},
			"defaultValue": "..."
		}
	],
	"groups": [
		{
			"title": "References",
			"kind": 16777216,
			"children": [
				7
			]
		},
		{
			"title": "Interfaces",
			"kind": 256,
			"children": [
				5
			]
		},
		{
			"title": "Type aliases",
			"kind": 4194304,
			"children": [
				4
			]
		},
		{
			"title": "Variables",
			"kind": 32,
			"children": [
				1
			]
		}
	]
}