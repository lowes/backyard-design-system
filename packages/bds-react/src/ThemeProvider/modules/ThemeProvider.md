# Namespace: ThemeProvider

## Variables

### bdsName

â€¢ **bdsName**: `string`
