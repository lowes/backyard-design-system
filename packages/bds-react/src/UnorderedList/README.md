# @lowes-tech/bds-react/UnorderedList

## Interfaces

- [UnorderedListProps](interfaces/UnorderedListProps.md)

## References

### default

Renames and re-exports [UnorderedList](README.md#unorderedlist)

## Type aliases

### UnorderedListRef

Ƭ **UnorderedListRef**: `HTMLUListElement`

## Variables

### UnorderedList

• **UnorderedList**: `BDSForwardRef`<[`UnorderedListProps`](interfaces/UnorderedListProps.md)\>
