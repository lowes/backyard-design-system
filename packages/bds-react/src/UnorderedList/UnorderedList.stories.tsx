import React from 'react'

import { withKnobs, select, boolean, number, text } from '@storybook/addon-knobs'

import { GridV3 as Grid, UnorderedList, ListItem, UnorderedListSkeleton } from '../'
import { ApiLink } from '../utils/storybook/ApiLink'

export default { title: '@lowes-tech/bds-react/UnorderedList', decorators: [withKnobs] }

const densities = [
    'Density',
    {
        condensed: 'condensed',
        normal: 'normal',
        comfort: 'comfort',
    },
    'normal',
] as const

export const Default = () => (
    <Grid.Column start={6} end={12}>
        <UnorderedList density={select(...densities)}>
            <ListItem>Lorem ipsum dolor sit amet, consectetur adipiscing.</ListItem>
            <ListItem>Lorem ipsum dolor sit amet, consectetur adipiscing.</ListItem>
            <ListItem>Lorem ipsum dolor sit amet, consectetur adipiscing.</ListItem>
            <ListItem>Lorem ipsum dolor sit amet, consectetur adipiscing.</ListItem>
        </UnorderedList>

        <ApiLink to='#' />
    </Grid.Column>
)

export const Nested = () => (
    <Grid.Column start={6} end={12}>
        <UnorderedList density={select(...densities)}>
            <ListItem>Lorem ipsum dolor sit amet, consectetur adipiscing.</ListItem>
            <ListItem>Lorem ipsum dolor sit amet, consectetur adipiscing.</ListItem>
            <ListItem>
                Lorem ipsum dolor sit amet, consectetur adipiscing.
                <UnorderedList>
                    <ListItem>Look at me, I'm nested and a slightly different color.</ListItem>
                    <ListItem>
                        Lorem ipsum dolor sit amet, consectetur adipiscing.
                        <UnorderedList>
                            <ListItem>Lorem ipsum dolor sit amet, consectetur adipiscing.</ListItem>
                            <ListItem>Lorem ipsum dolor sit amet, consectetur adipiscing.</ListItem>
                        </UnorderedList>
                    </ListItem>
                </UnorderedList>
            </ListItem>
            <ListItem>Lorem ipsum dolor sit amet, consectetur adipiscing.</ListItem>
        </UnorderedList>

        <ApiLink to='#' />
    </Grid.Column>
)

export const Skeleton = () => (
    <Grid.Column start={6} end={12}>
        <UnorderedListSkeleton
            density={select(...densities)}
            items={number('List items', 4)}
            width={text('Width', '18.75rem')}
            animate={boolean('Animate', false)}
        />
    </Grid.Column>
)