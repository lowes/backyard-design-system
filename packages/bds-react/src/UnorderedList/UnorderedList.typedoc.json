{
	"id": 0,
	"name": "@lowes-tech/bds-react/UnorderedList",
	"kind": 1,
	"kindString": "Project",
	"flags": {},
	"children": [
		{
			"id": 6,
			"name": "default",
			"kind": 16777216,
			"kindString": "Reference",
			"flags": {},
			"target": 1
		},
		{
			"id": 2,
			"name": "UnorderedListProps",
			"kind": 256,
			"kindString": "Interface",
			"flags": {},
			"children": [
				{
					"id": 4,
					"name": "as",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"type": {
						"type": "reference",
						"typeArguments": [
							{
								"type": "intrinsic",
								"name": "any"
							}
						],
						"name": "ElementType"
					},
					"inheritedFrom": {
						"type": "reference",
						"name": "BackyardBaseProps.as"
					}
				},
				{
					"id": 3,
					"name": "density",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"type": {
						"type": "union",
						"types": [
							{
								"type": "literal",
								"value": "normal"
							},
							{
								"type": "literal",
								"value": "comfort"
							},
							{
								"type": "literal",
								"value": "condensed"
							}
						]
					}
				}
			],
			"groups": [
				{
					"title": "Properties",
					"kind": 1024,
					"children": [
						4,
						3
					]
				}
			],
			"extendedTypes": [
				{
					"type": "reference",
					"typeArguments": [
						{
							"type": "reference",
							"id": 5,
							"name": "UnorderedListRef"
						}
					],
					"name": "BackyardBaseProps"
				}
			]
		},
		{
			"id": 5,
			"name": "UnorderedListRef",
			"kind": 4194304,
			"kindString": "Type alias",
			"flags": {},
			"type": {
				"type": "reference",
				"name": "HTMLUListElement"
			}
		},
		{
			"id": 1,
			"name": "UnorderedList",
			"kind": 32,
			"kindString": "Variable",
			"flags": {},
			"type": {
				"type": "reference",
				"typeArguments": [
					{
						"type": "reference",
						"id": 2,
						"name": "UnorderedListProps"
					}
				],
				"name": "BDSForwardRef"
			},
			"defaultValue": "..."
		}
	],
	"groups": [
		{
			"title": "References",
			"kind": 16777216,
			"children": [
				6
			]
		},
		{
			"title": "Interfaces",
			"kind": 256,
			"children": [
				2
			]
		},
		{
			"title": "Type aliases",
			"kind": 4194304,
			"children": [
				5
			]
		},
		{
			"title": "Variables",
			"kind": 32,
			"children": [
				1
			]
		}
	]
}