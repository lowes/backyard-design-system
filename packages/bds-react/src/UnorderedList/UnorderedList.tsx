import * as React from 'react'
import classNames from 'classnames'

import type { BackyardBaseProps } from '../utils/typings/BackyardProps'
import type { BDSForwardRef } from '../utils/typings/BDSComponentProps'
import { useBackyardTheme, validateBackyardTheme } from '../ThemeProvider'

import { UnorderedListWrapper } from './styles/UnorderedListWrapper'

const UnorderedList: BDSForwardRef<UnorderedListProps> = React.forwardRef<
    UnorderedListRef,
    UnorderedListProps
>(function UnorderedList({ children, density, className, ...props }, ref) {
    // Get Backyard Theme Context
    const theme = useBackyardTheme()

    validateBackyardTheme(theme, 'UnorderedList')

    return (
        <UnorderedListWrapper
            className={classNames(density ? `density--${density}` : null, className)}
            ref={ref}
            {...props}
        >
            {children}
        </UnorderedListWrapper>
    )
})

type UnorderedListRef = HTMLUListElement

interface UnorderedListProps extends BackyardBaseProps<UnorderedListRef> {
    density?: 'normal' | 'comfort' | 'condensed'
}

UnorderedList.bdsName = 'UnorderedList'

export { UnorderedList }

export type { UnorderedListProps, UnorderedListRef }

export default UnorderedList
