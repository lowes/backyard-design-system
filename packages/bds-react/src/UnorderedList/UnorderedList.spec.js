import * as React from 'react'
import { render } from '../../test-utils'
import { ThemeProvider } from '../ThemeProvider'
import UnorderedList from './UnorderedList'
import ListItem from '../ListItem'

describe('UnorderedList Snapshots', () => {
    test('UnorderdList default snapshot', () => {
        const { asFragment } = render(
            <UnorderedList density='normal'>
                <ListItem>List item</ListItem>
            </UnorderedList>
        )

        expect(asFragment()).toMatchSnapshot()
    })

    describe('themes', () => {
        const themes = ['light', 'dark']
        const densities = ['condensed', 'normal', 'comfort']

        themes.forEach((theme) => {
            densities.forEach((density) => {
                test(`UnorderedList normal spacing ${theme} theme snapshot`, () => {
                    const { asFragment } = render(
                        <ThemeProvider theme={theme}>
                            <React.Fragment>
                                <UnorderedList density={density}>
                                    <ListItem>Lorem ipsum dolor sit amet, consectetur adipiscing.</ListItem>
                                    <ListItem>Lorem ipsum dolor sit amet, consectetur adipiscing.</ListItem>
                                    <ListItem>Lorem ipsum dolor sit amet, consectetur adipiscing.
                                        <UnorderedList nested>
                                            <ListItem>Look at me, I'm nested and a slightly different color.</ListItem>
                                            <ListItem>Lorem ipsum dolor sit amet, consectetur adipiscing.</ListItem>
                                        </UnorderedList>
                                    </ListItem>
                                    <ListItem>Lorem ipsum dolor sit amet, consectetur adipiscing.</ListItem>
                                </UnorderedList>
                            </React.Fragment>
                        </ThemeProvider>
                    )
                    expect(asFragment()).toMatchSnapshot()
                })
            })
        })
    })
})