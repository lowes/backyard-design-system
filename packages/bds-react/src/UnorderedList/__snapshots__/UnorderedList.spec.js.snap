// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnorderedList Snapshots UnorderdList default snapshot 1`] = `
<DocumentFragment>
  .c0 {
  list-style-type: disc;
  color: var(--bds-color-text-primary);
  padding-left: var(--bds-sizes-size-16);
}

.c0 li {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: var(--bds-font-weight-regular);
  font-size: var(--bds-sizes-size-16);
  line-height: var(--bds-sizes-size-24);
}

.c0 ul {
  padding-left: var(--bds-sizes-size-16);
}

.c0.density--condensed li {
  padding-top: var(--bds-sizes-size-4);
}

.c0.density--normal li {
  padding-top: var(--bds-sizes-size-8);
}

.c0.density--comfort li {
  padding-top: var(--bds-sizes-size-12);
}

.c1 {
  font-size: var(--bds-sizes-size-14);
  line-height: var(--bds-sizes-size-20);
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

@media only screen and (min-width:82rem) {
  .c1 {
    font-size: var(--bds-sizes-size-16);
    line-height: var(--bds-sizes-size-24);
  }
}

<ul
    class="c0 density--normal"
  >
    <li
      class="c1"
    >
      List item
    </li>
  </ul>
</DocumentFragment>
`;

exports[`UnorderedList Snapshots themes UnorderedList normal spacing dark theme snapshot 1`] = `
<DocumentFragment>
  .c0 {
  list-style-type: disc;
  color: var(--bds-color-text-primary);
  padding-left: var(--bds-sizes-size-16);
}

.c0 li {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: var(--bds-font-weight-regular);
  font-size: var(--bds-sizes-size-16);
  line-height: var(--bds-sizes-size-24);
}

.c0 ul {
  padding-left: var(--bds-sizes-size-16);
}

.c0.density--condensed li {
  padding-top: var(--bds-sizes-size-4);
}

.c0.density--normal li {
  padding-top: var(--bds-sizes-size-8);
}

.c0.density--comfort li {
  padding-top: var(--bds-sizes-size-12);
}

.c1 {
  font-size: var(--bds-sizes-size-14);
  line-height: var(--bds-sizes-size-20);
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

@media only screen and (min-width:82rem) {
  .c1 {
    font-size: var(--bds-sizes-size-16);
    line-height: var(--bds-sizes-size-24);
  }
}

<ul
    class="c0 density--condensed"
  >
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
    </li>
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
    </li>
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
      <ul
        class="c0"
      >
        <li
          class="c1"
        >
          Look at me, I'm nested and a slightly different color.
        </li>
        <li
          class="c1"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing.
        </li>
      </ul>
    </li>
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
    </li>
  </ul>
</DocumentFragment>
`;

exports[`UnorderedList Snapshots themes UnorderedList normal spacing dark theme snapshot 2`] = `
<DocumentFragment>
  .c0 {
  list-style-type: disc;
  color: var(--bds-color-text-primary);
  padding-left: var(--bds-sizes-size-16);
}

.c0 li {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: var(--bds-font-weight-regular);
  font-size: var(--bds-sizes-size-16);
  line-height: var(--bds-sizes-size-24);
}

.c0 ul {
  padding-left: var(--bds-sizes-size-16);
}

.c0.density--condensed li {
  padding-top: var(--bds-sizes-size-4);
}

.c0.density--normal li {
  padding-top: var(--bds-sizes-size-8);
}

.c0.density--comfort li {
  padding-top: var(--bds-sizes-size-12);
}

.c1 {
  font-size: var(--bds-sizes-size-14);
  line-height: var(--bds-sizes-size-20);
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

@media only screen and (min-width:82rem) {
  .c1 {
    font-size: var(--bds-sizes-size-16);
    line-height: var(--bds-sizes-size-24);
  }
}

<ul
    class="c0 density--normal"
  >
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
    </li>
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
    </li>
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
      <ul
        class="c0"
      >
        <li
          class="c1"
        >
          Look at me, I'm nested and a slightly different color.
        </li>
        <li
          class="c1"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing.
        </li>
      </ul>
    </li>
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
    </li>
  </ul>
</DocumentFragment>
`;

exports[`UnorderedList Snapshots themes UnorderedList normal spacing dark theme snapshot 3`] = `
<DocumentFragment>
  .c0 {
  list-style-type: disc;
  color: var(--bds-color-text-primary);
  padding-left: var(--bds-sizes-size-16);
}

.c0 li {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: var(--bds-font-weight-regular);
  font-size: var(--bds-sizes-size-16);
  line-height: var(--bds-sizes-size-24);
}

.c0 ul {
  padding-left: var(--bds-sizes-size-16);
}

.c0.density--condensed li {
  padding-top: var(--bds-sizes-size-4);
}

.c0.density--normal li {
  padding-top: var(--bds-sizes-size-8);
}

.c0.density--comfort li {
  padding-top: var(--bds-sizes-size-12);
}

.c1 {
  font-size: var(--bds-sizes-size-14);
  line-height: var(--bds-sizes-size-20);
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

@media only screen and (min-width:82rem) {
  .c1 {
    font-size: var(--bds-sizes-size-16);
    line-height: var(--bds-sizes-size-24);
  }
}

<ul
    class="c0 density--comfort"
  >
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
    </li>
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
    </li>
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
      <ul
        class="c0"
      >
        <li
          class="c1"
        >
          Look at me, I'm nested and a slightly different color.
        </li>
        <li
          class="c1"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing.
        </li>
      </ul>
    </li>
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
    </li>
  </ul>
</DocumentFragment>
`;

exports[`UnorderedList Snapshots themes UnorderedList normal spacing light theme snapshot 1`] = `
<DocumentFragment>
  .c0 {
  list-style-type: disc;
  color: var(--bds-color-text-primary);
  padding-left: var(--bds-sizes-size-16);
}

.c0 li {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: var(--bds-font-weight-regular);
  font-size: var(--bds-sizes-size-16);
  line-height: var(--bds-sizes-size-24);
}

.c0 ul {
  padding-left: var(--bds-sizes-size-16);
}

.c0.density--condensed li {
  padding-top: var(--bds-sizes-size-4);
}

.c0.density--normal li {
  padding-top: var(--bds-sizes-size-8);
}

.c0.density--comfort li {
  padding-top: var(--bds-sizes-size-12);
}

.c1 {
  font-size: var(--bds-sizes-size-14);
  line-height: var(--bds-sizes-size-20);
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

@media only screen and (min-width:82rem) {
  .c1 {
    font-size: var(--bds-sizes-size-16);
    line-height: var(--bds-sizes-size-24);
  }
}

<ul
    class="c0 density--condensed"
  >
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
    </li>
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
    </li>
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
      <ul
        class="c0"
      >
        <li
          class="c1"
        >
          Look at me, I'm nested and a slightly different color.
        </li>
        <li
          class="c1"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing.
        </li>
      </ul>
    </li>
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
    </li>
  </ul>
</DocumentFragment>
`;

exports[`UnorderedList Snapshots themes UnorderedList normal spacing light theme snapshot 2`] = `
<DocumentFragment>
  .c0 {
  list-style-type: disc;
  color: var(--bds-color-text-primary);
  padding-left: var(--bds-sizes-size-16);
}

.c0 li {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: var(--bds-font-weight-regular);
  font-size: var(--bds-sizes-size-16);
  line-height: var(--bds-sizes-size-24);
}

.c0 ul {
  padding-left: var(--bds-sizes-size-16);
}

.c0.density--condensed li {
  padding-top: var(--bds-sizes-size-4);
}

.c0.density--normal li {
  padding-top: var(--bds-sizes-size-8);
}

.c0.density--comfort li {
  padding-top: var(--bds-sizes-size-12);
}

.c1 {
  font-size: var(--bds-sizes-size-14);
  line-height: var(--bds-sizes-size-20);
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

@media only screen and (min-width:82rem) {
  .c1 {
    font-size: var(--bds-sizes-size-16);
    line-height: var(--bds-sizes-size-24);
  }
}

<ul
    class="c0 density--normal"
  >
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
    </li>
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
    </li>
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
      <ul
        class="c0"
      >
        <li
          class="c1"
        >
          Look at me, I'm nested and a slightly different color.
        </li>
        <li
          class="c1"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing.
        </li>
      </ul>
    </li>
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
    </li>
  </ul>
</DocumentFragment>
`;

exports[`UnorderedList Snapshots themes UnorderedList normal spacing light theme snapshot 3`] = `
<DocumentFragment>
  .c0 {
  list-style-type: disc;
  color: var(--bds-color-text-primary);
  padding-left: var(--bds-sizes-size-16);
}

.c0 li {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: var(--bds-font-weight-regular);
  font-size: var(--bds-sizes-size-16);
  line-height: var(--bds-sizes-size-24);
}

.c0 ul {
  padding-left: var(--bds-sizes-size-16);
}

.c0.density--condensed li {
  padding-top: var(--bds-sizes-size-4);
}

.c0.density--normal li {
  padding-top: var(--bds-sizes-size-8);
}

.c0.density--comfort li {
  padding-top: var(--bds-sizes-size-12);
}

.c1 {
  font-size: var(--bds-sizes-size-14);
  line-height: var(--bds-sizes-size-20);
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

@media only screen and (min-width:82rem) {
  .c1 {
    font-size: var(--bds-sizes-size-16);
    line-height: var(--bds-sizes-size-24);
  }
}

<ul
    class="c0 density--comfort"
  >
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
    </li>
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
    </li>
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
      <ul
        class="c0"
      >
        <li
          class="c1"
        >
          Look at me, I'm nested and a slightly different color.
        </li>
        <li
          class="c1"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing.
        </li>
      </ul>
    </li>
    <li
      class="c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing.
    </li>
  </ul>
</DocumentFragment>
`;
